### DISCUSSION TITLE
New install script for simplified Docker image deployment

### DISCUSSION THREAD

**Discussion by tphakala** [tphakala/birdnet-go#325] [Announcements]
I have created an install.sh script to simplify BirdNET-Go deployment as a Docker container. The installation script is currently supported only on apt-based Linux distributions (including Debian, Ubuntu, and Raspberry Pi OS).

Please test if you are able to and provide feedback, thanks!

```
curl -fsSL https://github.com/tphakala/birdnet-go/raw/main/install.sh -o install.sh
bash ./install.sh
```

**Comment by mixylplik3** [comment]
Thank you. This worked perfectly. Installed in a fresh Docker LXC as installed by the script from Proxmox Helper-Scripts. I've been struggling to get any version of BirdNET working reliably in a virtualized environment until now.

**Comment by JamesW-AU** [comment]
@mixylplik3 Did you have to do anything special preparing your LXC? I have tried both unprivileged and privileged LXC through the Docker helper-script and in both cases I get this error:

[Install]
WantedBy=multi-user.target
Created symlink /etc/systemd/system/multi-user.target.wants/birdnet-go.service → /etc/systemd/system/birdnet-go.service.

🚀 Starting BirdNET-Go...
❌ Failed to start BirdNET-Go service

📋 Service logs (last 20 entries):
Jun 02 22:52:22 docker-priv systemd[1]: /etc/systemd/system/birdnet-go.service:25: Missing '=', ignoring line.
Jun 02 22:52:22 docker-priv systemd[1]: Starting birdnet-go.service - BirdNET-Go...
Jun 02 22:52:22 docker-priv docker[2180]: Error response from daemon: No such container: birdnet-go
Jun 02 22:52:22 docker-priv systemd[1]: Started birdnet-go.service - BirdNET-Go.
Jun 02 22:52:22 docker-priv docker[2190]: docker: 'docker run' requires at least 1 argument
Jun 02 22:52:22 docker-priv docker[2190]: Usage:  docker run [OPTIONS] IMAGE [COMMAND] [ARG...]
Jun 02 22:52:22 docker-priv docker[2190]: See 'docker run --help' for more information
Jun 02 22:52:22 docker-priv systemd[1]: birdnet-go.service: Main process exited, code=exited, status=1/FAILURE
Jun 02 22:52:22 docker-priv sh[2201]: umount: /root/birdnet-go-app/config/hls: block devices are not permitted on filesystem.

**Comment by AdamM68** [comment]
is there any new script for docker installation ?
I plan to install birdnetGO on ma HP T640 (Linux Mint 22.3) in a docker - via Portainer. I wonder if there is any new suggestions how to do this easy ?
Thanks for help.

PS - I have birdnetPi running for nearly 2 years on rPi4 - I want to compare this to BirdnetGO

**Reply by Aimache** [reply]
If you use Portainer, better to use the docker-compose file you can find here : doc/wiki/docker_compose_guide.md

**Reply by AdamM68** [reply]
done ! works perfect, instalation via portainer stacks
